src/httpclient.c

```diff
@@ -583,8 +586,9 @@ int main(int argc, char **argv)
             }
             break;
         case 'x':
-            if (h2o_url_init(&req.connect_to, NULL, h2o_iovec_init(optarg, strlen(optarg)), h2o_iovec_init(NULL, 0)) != 0 ||
-                req.connect_to._port == 0 || req.connect_to._port == 65535) {
+            req.connect_to = h2o_mem_alloc(sizeof(*req.connect_to));
+            if (h2o_url_init(req.connect_to, NULL, h2o_iovec_init(optarg, strlen(optarg)), h2o_iovec_init(NULL, 0)) != 0 ||
+                req.connect_to->_port == 0 || req.connect_to->_port == 65535) {
                 fprintf(stderr, "invalid server address specified for -X\n");
```

Comment:
sorry for a side-tracked request, but can you add a patch
to fix this message as well? I think this must be -x instead of -X.